ci/infra/libvirt/worker-instance.tf

```diff
@@ -101,25 +101,37 @@ resource "libvirt_domain" "worker" {
   }
 }
 
-resource "null_resource" "worker_wait_cloudinit" {
+resource "null_resource" "worker" {
   depends_on = [libvirt_domain.worker]
   count      = var.workers
 
+  triggers = {
+    worker_ips = "${join(",", libvirt_domain.worker.*.network_interface.0.addresses.0)}"
+    username   = var.username
+  }
+
   connection {
     host = element(
-      libvirt_domain.worker.*.network_interface.0.addresses.0,
-      count.index
+      split(",", self.triggers.worker_ips),
+      count.index,
     )
-    user     = var.username
-    password = var.password
-    type     = "ssh"
+    user  = self.triggers.username
+    type  = "ssh"
+    agent = true
   }
 
   provisioner "remote-exec" {
     inline = [
       "cloud-init status --wait > /dev/null",
     ]
   }
+
+  provisioner "remote-exec" {
+    when = destroy
+    inline = [
+      "if sudo SUSEConnect -s | grep -v 'Not Registered' > /dev/null 2>&1; then sudo SUSEConnect -d; fi"
+    ]
+  }
 }
 
 resource "null_resource" "worker_reboot" {
```

Comment:
this resource uses old null_resource name, it should use:
`depends_on = [null_resource.worker]`

+ the same for master instance

Reply:
also `provisioner "local-exec"` bellow uses the old way for getting host address for connection

Reply:
> this resource uses old null_resource name, it should use:
> `depends_on = [null_resource.worker]`
> 
>     * the same for master instance

yes good catch, I've pushed the changes for `libvirt` and `openstack`